db/migrate/20171105193838_add_timestamps_to_competitors.rb

```diff
@@ -0,0 +1,6 @@
+class AddTimestampsToCompetitors < ActiveRecord::Migration[5.1]
+  def change
+    add_column :competitors, :created_at, :datetime, null: false, default: Time.now
```

Comment:
Is datetime a known  global or where is it defined? (Is this the appropriate place to ask this question?)

Reply:
This is the perfect place to ask questions! Gives me lots o context. This answer will probably be a lot more info than necessary.

In this call to change, we are defining how the database should change after a migration is run. If we migration forward, it will add the column, if we migrate backwards it will remove it.

:datetime in this scenario is a symbol, its an argument being based to the ```ActiveRecord::Migration``` function ```add_column```.

The rest of the arguments tell us:

Add a column to the ```competitors``` table, called ```created_at```. It should be a ```datetime```, non-null, and the default should be right now.

The full list of arguments for add_column can be found [in the docs](https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/add_column). The full list of supported datatypes (what datetime is) is dependent on the database.

We use sqlite in development, which has the most restrictive set of datatypes. [The full list is here.](https://stackoverflow.com/questions/17918117/rails-4-list-of-available-datatypes)

Reply:
That was the perfect amount of info! Seems that a majority of my learning needs is what functions/methods and globals (probably not the correct nomenclature) exist using ruby on rails, or any framework for that matter.

Reply:
Fortunately, all of that stuff is easy to find, and lots of editors have that information built right in, so you don't have to memorize (yay!). Remembering this kind of stuff off hand just comes from doing it over and over.

The harder part is knowing when its appropriate to use each thing. Sometimes one data structure makes sense for one approach, and other times it would be really inefficient.